### ISSUE TITLE
False positive in CakePHP.Commenting.FunctionComment.InvalidReturn

### ISSUE THREAD

**Issue by bancer** [cakephp/cakephp-codesniffer#311]
In a function doc block:
```
 * @return (int|string)[]|(int|string)[][]
```

Reported:
```
 2331 | ERROR | Expected "(int|string)[]|string)[][]" but found "(int|string)[]|(int|string)[][]" for function return type
      |       | (CakePHP.Commenting.FunctionComment.InvalidReturn)
```
The string expected by the sniff seems to be invalid: "(int|string)[]|string)[][]". There is one opening brace but two closing.

**Comment by othercorey** [comment]
Which version of cakephp-cs are you using?

**Comment by bancer** [comment]
The latest version according to the composer.json: `"cakephp/cakephp-codesniffer": "^4.0.0",`.
